propagation_ot.go

```diff
@@ -85,18 +83,24 @@ func (p *textMapPropagator) Extract(
 		err          error
 	)
 	decodedBaggage := make(map[string]string)
+
+	var traceIDFound, spanIDFound bool
```

Comment:
Used this approach rather than a counter because it is more robust.

Because the case normalization happens inside the `ForeachKey` function, in the previous code, if the carrier stored multiple keys in different cases (for example, `X-B3-TraceId` and `X-B3-Traceid`), then the counter would get incremented twice even though only one field was actually set.